### DISCUSSION TITLE
Run workflow after some checks have completed

### DISCUSSION THREAD

**Discussion by pantelis-karamolegkos** [community/community#34452] [Actions]
I have a few branches that are not `master` nor `main` and none of them is the `default` branch.

Each commit on those branches creates some checks (3 actually).

I want to trigger a workflow once these checks have completed, and to be able to retrieve the result of each one of those checks.

I am aware of the [`check_run`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#check_run) and [`check_suite`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#check_run) triggers, but the problem is that according to the documentation the workflow file should be in the default branch.

How can I overcome this limitation?

Does the [`check_suite`](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#check_run) execute when **all checks** have completed? (didn't quite get the semantics by reading the docs)

**Comment by antonblr** [comment]
>Does the [check_suite](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#check_run) execute when all checks have completed?

Yes, a check suite is a collection of the check runs created for a specific commit. Check suites summarize the status and conclusion of the check runs that are in the suite.

>How can I overcome this limitation?
Why can't you keep your final workflow in the default branch and wait for `check_suite` event from non-default branches?

**Comment by pantelis-karamolegkos** [comment]
@antonblr thanks for the feedback.

To group checks in a `check_suite` do I have to do an explicit action?

Or do all checks that are being added get automatically grouped in a `check_suite`?

**Comment by antonblr** [comment]
All checks created by specific commit are grouped to check suite automatically. You can see all check runs belonging to a given check suite with this API call: 
```
curl \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer <YOUR-TOKEN>" \
  https://api.github.com/repos/OWNER/REPO/check-suites/CHECK_SUITE_ID/check-runs
```

To get `CHECK_SUITE_ID` - list all check suits for a given REF and pick whatever you want to verify:
```
curl \
  -H "Accept: application/vnd.github+json" \
  -H "Authorization: Bearer <YOUR-TOKEN>" \
  https://api.github.com/repos/OWNER/REPO/commits/REF/check-suites
```
